### ISSUE TITLE
False positive from `constructor-super`

### ISSUE THREAD

**Issue by zaygraveyard** [eslint/eslint#5394]
##### The version of ESLint you are using.
```
$ ./node_modules/.bin/eslint -v
v2.2.0
```

##### What you did.
```js
class A extends Object {
  constructor() {
    super();
    for (let i = 0; i < 0; i++);
  }
}
```

##### What you expected to happen.

The code passes the linting

##### What actually happened.

eslint gave an error.

##### Your configuration

```json
{
  "rules": {
    "constructor-super": 2
  },
  "env": {
    "es6": true
  }
}
```

##### The complete and actual output from ESLint

```
$ eslint test.js
/tmp/eslint-bugs/test.js
  2:3  error  Lacked a call of 'super()' in some code paths  constructor-super

✖ 1 problem (1 error, 0 warnings)
```

**Comment by mysticatea** [comment]
Thank you for this issue.

I guess `"What you expected to happen."` and `"What actually happened."` are reversed in your intention.
I'm investigating....

**Comment by zaygraveyard** [comment]
@mysticatea Yep, sorry about that, I corrected it
